packages/next/client/components/navigation.ts

```diff
@@ -97,8 +152,19 @@ export function useRouter(): import('../../shared/lib/app-router-context').AppRo
 /**
  * Get the current pathname. For example usePathname() on /dashboard?foo=bar would return "/dashboard"
  */
-export function usePathname(): string {
-  return useContext(PathnameContext)
+export function usePathname(): string | null {
```

Comment:
Maybe this should throw in the `null` case. I'd prefer it to always be a `string` as that's how it's supposed to be in the new router.

Reply:
If we change it to throws then during migration, this would cause any component using it from pages to throw too. This sort of defeats the purpose for the hybrid approach.